src/core/worker.js

```diff
@@ -661,28 +664,32 @@ class WorkerMessageHandler {
             newXrefInfo = {
               rootRef: xref.trailer.getRaw("Root") || null,
               encryptRef: xref.trailer.getRaw("Encrypt") || null,
-              newRef: xref.getNewRef(),
+              newRef: xref.getNewTemporaryRef(),
               infoRef: xref.trailer.getRaw("Info") || null,
               info: infoObj,
               fileIds: xref.trailer.get("ID") || null,
               startXRef,
               filename,
             };
           }
-          xref.resetNewRef();
 
-          return incrementalUpdate({
+          const data = incrementalUpdate({
```

Comment:
Can this function call ever throw, because if so we're not resetting the temporary Refs which could potentially cause issues later on?
Would it make sense to do something along these lines instead?
```suggestion
          try {
            const data = incrementalUpdate({ ... });
            xref.resetNewTemporaryRef();
            return data;
          } catch (reason) {
            xref.resetNewTemporaryRef();
            throw reason;
          }
```